### PR TITLE
Updated csr.yaml.m4 for easier script parsing, fixed duplicate entries

### PR DESCRIPTION
**Pull request by silabs-hfegran**
Changes: 
Fixed some entries that would be included twice.
Fixed typos
Merged register definitions for easier parsing

Reason for change: Adapt m4 format to be able to generate bitfield data types for c/systemverilog
Changes that are not bugfixes should have no functional impact on current infrastructure

### CODE REVIEW

**Comment by silabs-hfegran** [comment]
I will likely need to make some more small modifications, please do not merge yet.

File: yaml/csr.yaml.m4 [hunk 1]
```diff
@@ -2661,6 +2670,7 @@ ifelse(eval(X_EXT != 0), 1, [[[
       lsb: 21
       warl_legalize: |
         val_out = 0
+]]])
```
Comment (Silabs-ArjanB) [inline yaml/csr.yaml.m4:2673]: This clause is okay, but the next one is wrong. If V_EXT != 0, then the V bit will be hardwired to 1 (so WARL (0x1))

File: yaml/csr.yaml.m4 [hunk 2]
```diff
@@ -2730,6 +2741,7 @@ ifelse(eval(UMODE != 0), 1, [[[
       lsb: 15
       warl_legalize: |
         val_out = 0
+]]])
```
Comment (Silabs-ArjanB) [inline yaml/csr.yaml.m4:2744]: Same here. This clause is okay, but the next one is wrong. If P_EXT != 0, then the P bit will be hardwired to 1 (so WARL (0x1))

File: yaml/csr.yaml.m4 [hunk 3]
```diff
@@ -2901,29 +2913,32 @@ ifelse(eval(I_BASE != 0), 1, [[[
       type: WARL
       msb: 0
       lsb: 0
+ifelse(eval(A_EXT == 0), 1, [[[
```
Comment (Silabs-ArjanB) [inline yaml/csr.yaml.m4:2916]: A_EXT is no longer a single bit, but can be A_NONE, ZALRSC, A. Only if set to 'A' should the misa.A bit be set to 1. check all uses of A_EXT in this file.

File: yaml/csr.yaml.m4 [hunk 4]
```diff
@@ -4091,47 +4113,53 @@ ifelse(eval(NUM_MHPMCOUNTERS >= 29), 1, [[[
...
+]]])
+ifelse(eval(UMODE == 0), 1, [[[
       warl_legalize: |
         val_out = val_in if ((val_in == 0x3) or (val_in == 0x0)) else val_orig
 ]]])
+      subtype: clic
     - field_name: MPIE
       description: >
-          Alias for mstatus.MPIE
+          Aliaaas for mstatus.MPIE
```
Comment (Silabs-ArjanB) [inline yaml/csr.yaml.m4:4157]: Aliaaas -> Alias

**Review by Silabs-ArjanB — CHANGES_REQUESTED** [review]
Changes requested as commented.

This PR should be done on CV32E40X instead though.

**Comment by Silabs-ArjanB** [comment]
@silabs-hfegran What needs to be done with this PR (except for resolving the merge conflicts)? If it is still relevant when can it be merged?

**Pull request closed without merging**